core/src/executor/alter/table.rs

```diff
@@ -146,8 +219,84 @@ pub async fn drop_table<T: GStore + GStoreMut>(
             schema.ok_or_else(|| AlterError::TableNotFound(table_name.to_owned()))?;
         }
 
+        let schemas = storage.fetch_all_schemas().await?;
+        let referring_children: Vec<ReferringChild> = schemas
+            .into_iter()
+            .flat_map(
+                |Schema {
+                     table_name: referring_table_name,
+                     foreign_keys,
+                     ..
+                 }| {
+                    foreign_keys
+                        .into_iter()
+                        .filter_map(
+                            |ForeignKey {
+                                 name: constraint_name,
+                                 referred_table,
+                                 ..
+                             }| {
+                                if &referred_table == table_name
+                                    && &referring_table_name != table_name
+                                {
+                                    return Some(ReferringChild {
+                                        table_name: referring_table_name.clone(),
+                                        constraint_name,
+                                    });
+                                }
+
+                                None
+                            },
+                        )
+                        .collect::<Vec<_>>()
+                },
+            )
+            .collect();
```

Comment:
```rust
let referring_children: Vec<ReferringChild> = schemas
    .into_iter()
    .flat_map(move |schema| {
        let Schema {
            table_name: referring_table_name,
            foreign_keys,
            ..
        } = schema;

        foreign_keys.into_iter().filter_map(move |foreign_key| {
            let ForeignKey {
                name: constraint_name,
                referred_table,
                ..
            } = foreign_key;

            (&referred_table == table_name && &referring_table_name != table_name)
                .then_some(ReferringChild {
                    table_name: referring_table_name.clone(),
                    constraint_name,
                })
        })
    })
    .collect();
```

I think that we can simply the codes more on this, I will try. it looks possible to simplify the below codes, too.